pkgs/applications/backup/unifi-protect-backup/default.nix

```
@@ -0,0 +1,37 @@
+{ fetchFromGitHub, python3, lib }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "unifi-protect-backup";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "ep1cman";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-HAiyNFWLs1McrlAB48me/iI15LssO8ec7BiWuJbRlbs=";
+  };
+
+  preBuild = ''
+    sed -i 's_click = "8.0.1"_click = "^8"_' pyproject.toml
+    sed -i 's_pyunifiprotect = .*_pyunifiprotect = "*"_' pyproject.toml
+  '';
+
+  nativeBuildInputs = with python3.pkgs; [
+    poetry
```

```suggestion
    poetry-core
```

or else you're including the whole cli and package management shebang instead of just the builder.